api/v1alpha1/sink_webhook.go

```diff
@@ -49,9 +49,19 @@ var _ webhook.Defaulter = &Sink{}
 func (r *Sink) Default() {
 	sinklog.Info("default", "name", r.Name)
 
-	if r.Spec.Replicas == nil {
-		r.Spec.Replicas = new(int32)
-		*r.Spec.Replicas = 1
+	if !(r.Spec.Replicas != nil && r.Spec.MinReplicas != nil) {
+		if r.Spec.MinReplicas != nil && r.Spec.Replicas == nil {
+			r.Spec.Replicas = new(int32)
+			*r.Spec.Replicas = *r.Spec.MinReplicas
+		} else if r.Spec.MinReplicas == nil && r.Spec.Replicas != nil {
+			r.Spec.MinReplicas = new(int32)
+			*r.Spec.MinReplicas = *r.Spec.Replicas
+		} else {
+			r.Spec.Replicas = new(int32)
```

Comment:
will FM still have the HPA error since the `replicas` are set to 1 even user doesn't set it?

Reply:
In this PR, `spec.replicas` becomes optional, so that the user does not need to configure this value (as in the argo template), but only needs to configure a minimum value of `spec.minReplicas`.

If `spec.replicas` is changed under HPA control, applying the template file again (without `spec.replicas` configured in it) will not trigger reconciliation.